app/models/energy/ScheduleRuleset.py

```diff
@@ -0,0 +1,207 @@
+    mode: 'Mode'
+
+
+class ScheduleTypeLimits (BaseModel):
+    """Specifies the data types and limits for values contained in schedules."""
+
+    type: Enum('ScheduleTypeLimits', {'type': 'ScheduleTypeLimits'})
+
+    name: str = Schema(
+        ...,
+        regex=r'^[\s.A-Za-z0-9_-]*$',
+    )
+
+    lower_limit_value: float = Schema(
+        0,
+        description='Lower limit for the schedule type is entered.'
+    )
+
+    upper_limit_value: float = Schema(
+        1,
+        description='Upper limit for the schedule type is entered.'
+    )
+
+    numeric_type: NumericType
+
+    unit_type: UnitType
+
+
+class YesOrNo (str, Enum):
+    no = 'No'
+    yes = 'Yes'
+
+
+class ScheduleDay(BaseModel):
+    """Used to describe the daily schedule for a single simulation day."""
+    type: Enum('ScheduleDay', {'type': 'ScheduleDay'})
+
+    name: str = Schema(
+        ...,
+        regex=r'^[\s.A-Za-z0-9_-]*$',
+    )
+
+    interpolate_to_timestep: YesOrNo = YesOrNo.no
+
+    hour: int = Schema(
+        ...,
+        ge=0,
+        le=23,
+    )
+
+    minute: int = Schema(
+        ...,
+        ge=0,
+        le=59
+    )
+
+    value_until_time: float = Schema(
+        ...
+    )
+
+
+class ScheduleRuleset (BaseModel):
+    """Used to define a schedule for a default day, further described by ScheduleRule."""
+
+    type: Enum('ScheduleRuleset', {'type': 'ScheduleRuleset'})
+
+    name: str = Schema(
+        ...,
+        regex=r'^[\s.A-Za-z0-9_-]*$',
+    )
+
+    schedule_type_limits: ScheduleTypeLimits
+
+    default_day_schedule: List[ScheduleDay]
+
+    summer_designday_schedule: List[ScheduleDay]
+
+    winter_designday_schedule: List[ScheduleDay]
+
+
+class ScheduleRule(BaseModel):
+    """A set of rules assigned to schedule ruleset for specific periods of time and for
+  particular days of the week according to a priority sequence."""
+
+    type: Enum('ScheduleDay', {'type': 'ScheduleDay'})
+
+    name: str = Schema(
+        ...,
+        regex=r'^[\s.A-Za-z0-9_-]*$'
+    )
+
+    schedule_rule_set: List[ScheduleRuleset]
+
+    rule_order: int = Schema(
+        ...,
+        ge=0
+    )
+
+    schedule_day: List[ScheduleDay]
+
+    apply_sunday: YesOrNo = YesOrNo.no
+
+    apply_monday: YesOrNo = YesOrNo.no
+
+    apply_tuesday: YesOrNo = YesOrNo.no
+
+    apply_wednesday: YesOrNo = YesOrNo.no
+
+    apply_thursday: YesOrNo = YesOrNo.no
+
+    apply_friday: YesOrNo = YesOrNo.no
+
+    apply_saturday: YesOrNo = YesOrNo.no
+
+    apply_holiday: YesOrNo = YesOrNo.no
+
+    date_specification_type: DateSpecificationType = DateSpecificationType.date_range
+
+    start_month: int = Schema(
+        1,
+        ge=1,
+        le=12
+    )
+
+    start_day: int = Schema(
+        1,
+        ge=1,
+        le=31
+    )
+
+    end_month: int = Schema(
+        12,
+        ge=1,
+        le=12
+    )
+
+    end_day: int = Schema(
+        31,
+        ge=1,
+        le=31
+    )
+
+    specific_month: int = Schema(
```

Comment:
What is specific month?

Reply:
... and, along the same lines, what is `specific_day`?

Reply:
The openstudio IDD file allows the `Date Specification Type` to be either a `DateRange or `SpecificDates`. A schedule can be assigned to a list of specific dates using `SpecificDates` . The front end of OpenStudio however only allows a `DateRange` to be specified. 
Should I still keep the `SpecificDates` option in the schema? @chriswmackey @mostaphaRoudsari

Reply:
I don't think that we need SpecificDates in the schema.